dissect/cstruct/types/base.py

```diff
@@ -217,3 +217,4 @@ def _write(cls, stream: BinaryIO, data: list[Any]) -> int:
             raise ArraySizeError(f"Expected static array size {cls.num_entries}, got {actual_size} instead.")
 
         return cls.type._write_array(stream, data)
+BaseType.ArrayType = Array
```

Comment:
Could you add a `ArrayType = None` to the `BaseType` class with a short comment describing what it's for? Something along the lines of:

```
# Define the array type for this type, this defaults to the Array class below and is set after that's defined
ArrayType: ArrayMetaType = None
```